### DISCUSSION TITLE
How can I map :Git to :G?

### DISCUSSION THREAD

**Discussion by sdemura** [nvim-mini/mini.nvim#929] [General]
With vim-fugitive, `:G` is available for `:Git`, how can I set up the same with `mini.git`?

**Comment by echasnovski** [comment]
As far as I know, there doesn't appear to be an easy way to create a copy of an existing user command.

The best solution seems to be to create an command line abbreviation for `G` which would expand to `Git` after pressing space.

Following [this answer](https://stackoverflow.com/a/3879737), there are at least two approaches:

- Easy: `vim.cmd('cnoreabbrev G Git')`. I think this will be enough 99% of times. However, it will expand whenever a separate `G` is entered in a command line, including during search (like after `/`) or when it is not at the first column.
- More robust: `vim.cmd([[cnoreabbrev <expr> G ((getcmdtype() is# ':' && getcmdline() is# 'G')?('Git'):('G'))]])`. If you use Neovim 0.10, this can also be rewritten with `vim.keymap.set()`:

    ```lua
    local expand_g = function() return (vim.fn.getcmdtype() == ':' and vim.fn.getcmdline() == 'G') and 'Git' or 'G' end
    vim.keymap.set('ca', 'G', expand_g, { expr = true })
    ```

**Reply by sdemura** [reply]
This should be good enough fo rme. Thanks!
